sql/sql_show.cc

```diff
@@ -1923,7 +1923,11 @@ int store_create_info(THD *thd, TABLE_LIST *table_list, String *packet,
     if (table->s->encrypt_type.length)
     {
       packet->append(STRING_WITH_LEN(" ENCRYPTION="));
-      append_unescaped(packet, share->encrypt_type.str, share->encrypt_type.length);
+      append_unescaped(packet, table->s->encrypt_type.str, table->s->encrypt_type.length);
+    } else if (create_info.encrypt_type.length)
```

Comment:
Why do we need another branch here? Shouldn't `table->s->encrypt_type` be always populated with actual encryption type value?

Reply:
It's... similar, but not exactly the same. table->s is based on the sql parser. If the option was specified in the SQL query, it is in it.

But if a table is encrypted because of the variable, only the innodb flag is set, the information during the parsing is unchanged.

I found no good way to change it, but the `create_info` structure can be altered by the virtual `update_create_info` function, which I modified for innodb.

Reply:
Still not sure why we have unsynchronized data here. A handler for `SHOW CREATE TABLE` (`mysqld_show_create()`) before calling `store_create_info()` function calls `open_tables()` - a function which should populate all the table attributes (including encryption flag). This `open_tables()` is used all over the code (not only for `SHOW CREATE TABLE`) and if it doesn't do this, we have a problem.